### ISSUE TITLE
Validate that benches and tools crates are formatted in CI

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#10768]
Yep. This is what we have to do, otherwise someone else is gonna run into this eventually.

Additionally, from the Bevy side, this should be checked by CI.

_Originally posted by @rlidwka in https://github.com/bevyengine/bevy/pull/10758#pullrequestreview-1749866027_

**Comment by matiqo15** [comment]
The issue here lies in that benches are not checked by `cargo fmt --all -- --check`. I've found two ways to solve it:
1. We can add "benches" to members of workspace in [Cargo.toml](https://github.com/bevyengine/bevy/blob/main/Cargo.toml#L22). However, this could add additional overhead - especially as "benches" are currently explicitly [excluded from members](https://github.com/bevyengine/bevy/blob/main/Cargo.toml#L17).
2. We could add flag in [tools/ci/src/main.rs](https://github.com/bevyengine/bevy/blob/main/tools/ci/src/main.rs#L64). This would then be: `cmd!(sh, "cargo fmt --all --manifest-path  ../../../benches/Cargo.toml -- --check")`. While not best, it seems to be the only solution I've found if we don't want to add benches to workspace.

**Comment by alice-i-cecile** [comment]
I think we should start with solution 2 for now.

**Comment by mockersf** [comment]
The issue is that benches (and compile fail crates) are excluded from the workspace:
https://github.com/bevyengine/bevy/blob/4221f7e7e9ae5b2d664a067f1fbabc97a5806232/Cargo.toml#L15-L21

I don't remember why, but if we want the benches to be checked in CI they should be in the workspace

**Comment by BD103** [comment]
I fixed part of this issue in #16858 when I added benchmarks and `compile_fail` tests to the workspace. In #17330, however, new integration tests were added that are excluded from the workspace. @raldone01 is there a particular reason why you chose to exclude them?

**Comment by raldone01** [comment]
The reason the integration tests were excluded from the workspace was because their whole purpose is to be standalone crates that depend on bevy.

I think there was an issue with them also being part of the bevy workspace.

I did not want to add more complexity/magic to the integration tests by copy Cargo tomls and sk on. The integration tests work like regular crates when you cd into them.

**Comment by BD103** [comment]
Makes sense! I suppose being part of the workspace could affect their dependency resolution / inappropriately reuse build artifacts. Since it seems ideal to keep integration tests separate from the main workspace, I'll just modify the issue title to mention them.
